HW2/P4/driver.py

```diff
@@ -15,11 +15,66 @@
 from timer import Timer
 import threading
 
+import logging
+
+logging.basicConfig(level=logging.DEBUG,
+                    format='[%(levelname)s] (%(threadName)-10s) %(message)s',
+                    )
+
+
 def py_median_3x3(image, iterations=10, num_threads=1):
     ''' repeatedly filter with a 3x3 median '''
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
 
+    threads = []
+
+    # Create events that signals the threads to avoid collisions
+    events = np.array([threading.Event()]*(iterations*num_threads)).reshape([num_threads, iterations])
+
```

Comment:
Also did conditional variables, seems like they are closely related in the handling.